pkgs/by-name/dm/dmd/generic.nix

```
@@ -105,6 +105,7 @@ stdenv.mkDerivation (finalAttrs: {
   '' + lib.optionalString (lib.versionAtLeast version "2.089.0" && lib.versionOlder version "2.092.2") ''
     rm dmd/compiler/test/dshell/test6952.d
   '' + lib.optionalString (lib.versionAtLeast version "2.092.2") ''
+    rm dmd/compiler/test/fail_compilation/needspkgmod.d
```

```suggestion
    substituteInPlace dmd/compiler/test/fail_compilation/needspkgmod.d --replace-fail 'REQUIRED_ARGS: -Icompilable' 'REQUIRED_ARGS: -Icompilable -L--no-demangle'
```

I've tested the changes from this commit: https://github.com/dlang/dmd/pull/16833 and they worked.

I think it would be more appropriate to edit the test instead of removing it.

---

Sweet, I didn't know there was a patch fixing this test.